src/wallet/wallet.cpp

```diff
@@ -749,7 +749,7 @@ std::set<uint256> CWallet::GetConflicts(const uint256& txid) const
 bool CWallet::HasWalletSpend(const uint256& txid) const
 {
     AssertLockHeld(cs_wallet);
-    auto iter = mapTxSpends.lower_bound(COutPoint(txid, 0));
+    auto iter = mapTxSpends.find(COutPoint(txid, 0));
```

Comment:
Looks like a bug here. Previously it would return true if any output was spent, now it will only return true if the first output was spent. Could fix pretty easily by passing CTransaction& instead of txid, and looping over the outputs.

Reply:
Done

Reply:
> Looks like a bug here. Previously it would return true if any output was spent, now it will only return true if the first output was spent. Could fix pretty easily by passing const CTransaction& instead of txid, and looping over the outputs.

A suggestion for a followup: it could be useful to add a test for it to prevent regressions in the future.